### ISSUE TITLE
[B7] translates to Bb7

### ISSUE THREAD

**Issue by Beacha** [igrek51/android-songbook#6]
Thank you for this app, I am trying to use it and put some of my own custom songs in it. When I use a [B7] chord and ´detect chords´ followed by ´validate´ my [B7] chords are either not recognized or translated to Bb7 chord.

Further, not an issue but a request, could you incorporate a function to set the scroll speed (and start) per song?

**Comment by igrek51** [comment]
Hi, the issue you are facing may be related to different chords notations used across different countries and choosing the proper one. 

Let me explain few things first:
Chord "B7" is somehow ambiguous. it may be considered as 2 completely different chords depending on notation you have in mind. In European / German notation chord "B" is in fact American / English "Bb". American "B" is in turn referenced as "H" using European notation. So Without specifying notation, it is impossible to distinguish what author had in mind.

Therefore supporting both notations at once introduces a lot of complications when handling that and it is very important to indicate which notation you are using. 
Thus in Guitar Songbook you can choose which notation you are using when editing your own song (I bet you might have not chosen that and that was the reason of your issue). You can also choose the notation for displaying chords in settings.
Internally, Guitar Songbook keeps all songs in European notation. So that's the reason why you might see that transformed after editing, but dont worry, it transforms them back to your format (specified in settings) every time when you edit song again or when displaying it.

To sum up, you should check your settings and choose a proper chord notation - American / English in your case.
Also pay attention to selecting American chord notation when editing song (it all encourages me to simplify this process).

Let me know if that solved your problem.

**Comment by igrek51** [comment]
Regarding `detect chords` feature, it works only when chords are not wrapped in brackets yet. 
It will add the missing brackets then, but it works mostly for chords at the end of the line. 
In particular it would be misleading when trying to detect `Am` chord in the middle of sentence "I Am a horse".

**Comment by igrek51** [comment]
In terms of unpredictable converting to "Bb7", I guess I've finally found the reason of this bug. When importing a song from file, Guitar Songbook disregards chords notation and always tries to parse a file with European notation (and it thinks that actually it is a "Bb7" chord). After that, when you open a song for further editing, it converts the chords to your American notation, that's why "Bb7" appears.
Given that, I'll try to fix that soon. For now, I have a temporary workaround for you:
- Delete your custom song.
- Change your notation to "German / European" in settings
- Import your song from a text file.
- Open it for editing (Guitar Songbook will not convert it to "Bb7" because it thinks it is still European notation)
- When editing a song, change chord notation to "English".
- Leave editor and save a song (you will see that it converts chord to "H7", because it keeps songs internally using unified European notation, but don't worry it will convert it back each time you open it)
- Change your notation back to "English / American" in settings - then chords should be fine: editing as well as displaying song on preview.

**Comment by igrek51** [comment]
Disregarding chords notation when importing a song from a text file should not be an issue anymore. It has been solved by these commits:
- https://github.com/igrek51/android-songbook/commit/6cb3a6af597134fed523b07560ba3383651db064
- https://github.com/igrek51/android-songbook/commit/4afa531727b758d5cbd213483d78bdd9ca17e5f8

Basically this was done by supporting individual chords notations per each song and adding an option to convert chords from other notations. So now when importing a song, its content is being left intact. It's imported as it is without applying any conversions. 
Right now the song is being converted only when a song notation differs from the one chosen by user to be displayed.
